printer/util.js

```diff
@@ -0,0 +1,71 @@
+const snmp = require('net-snmp');
+
+/**
+ * inkLevel determines and returns the current ink level of a HP p2015dn printer
+ * @param {String} printer IP address of printer to query
+ * @returns {Promise} Promise with current ink level of the printer in % of capacity
+ */
+async function inkLevel(printer) {
+  const currentLevel = await getCurrentTonerLevel(printer);
+  const capacity = await getTonerCapacity(printer);
+
+  return (currentLevel / capacity) * 100;
+}
+
+/**
+ * getTonerCapacity makes snmp query against toner capacity OID and returns value
+ * @param {String} printer IP address of printer to query
+ * @returns {Promise} Promise with toner capacity in unknown units
+ */
+function getTonerCapacity(printer) {
+  return new Promise((resolve) => {
+    const session = snmp.createSession(printer, 'public');
+
+    const oids = ['1.3.6.1.2.1.43.11.1.1.8.1.1'];
```

Comment:
to add onto my last comment these oids should live inside of some enum so the values make a little more sense

```js
const SNMP_OBJECT_IDS = {
  TONER_CAPACITY: '1.3.6.1.2.1.43.11.1.1.8.1.1',
  CURRENT_TONER_LEVEL: '1.3.6.1.2.1.43.11.1.1.9.1.1',
};
```

with this being said we can then call on our abstracted snmp function like
```js
async function getTonerCapacity(printerIp) {
  await executeSnmpRequest(printerIp, [SNMP_OBJECT_IDS.TONER_CAPACITY]);
}
```

and then the world rejoyces at how cool we are